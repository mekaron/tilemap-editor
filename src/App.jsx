import React from 'react';
import NavigationBar from './components/NavigationBar';
import ToolButtons from './components/ToolButtons';
import UtilityControls from './components/UtilityControls';

export default function App() {
  return (
    <div id="tilemapjs_root" className="card tilemapjs_root">
      <a id="downloadAnchorElem" style={{ display: 'none' }}></a>
      <div className="nav_wrapper">
        <NavigationBar />
        <ToolButtons />
        <UtilityControls />
        <div>
          <button className="primary-button" id="confirmBtn">"apply"</button>
        </div>
      </div>
      <div id="layoutContainer" className="card_body"></div>
    </div>
  );
}
