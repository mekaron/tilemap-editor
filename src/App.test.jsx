import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

test('App renders menu items', () => {
  const { container } = render(<App />);
  expect(screen.getByText(/File/i)).toBeTruthy();
  expect(container.querySelector('.tileset_opt_field.header')).toMatchSnapshot();
});

test('App renders tool buttons', () => {
  const { container } = render(<App />);
  expect(screen.getByTitle('paint tiles')).toBeTruthy();
  expect(container.querySelector('#toolButtonsWrapper')).toMatchSnapshot();
});
