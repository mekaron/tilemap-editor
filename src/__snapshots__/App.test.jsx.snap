// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders menu items 1`] = `
<div
  class="tileset_opt_field header"
>
  <div
    class="menu file"
  >
    <span>
       File 
    </span>
    <div
      class="dropdown"
      id="fileMenuDropDown"
    >
      <a
        class="button item button-as-link"
        href="#popup2"
      >
        About
      </a>
      <div
        class="overlay"
        id="popup2"
      >
        <div
          class="popup"
        >
          <h4>
            Tilemap editor
          </h4>
          <a
            class="close"
            href="#"
          >
            ×
          </a>
          <div
            class="content"
          >
            <div>
              Created by Todor Imreorov (blurymind@gmail.com)
            </div>
            <br />
            <div>
              <a
                class="button-as-link"
                href="https://github.com/blurymind/tilemap-editor"
              >
                Project page (Github)
              </a>
            </div>
            <div>
              <a
                class="button-as-link"
                href="https://ko-fi.com/blurymind"
              >
                Donate page (ko-fi)
              </a>
            </div>
            <br />
            <div>
              Instructions:
            </div>
            <div>
              right click on map - picks tile
            </div>
            <div>
              mid-click - erases tile
            </div>
            <div>
              left-click adds tile
            </div>
            <div>
              right-click on tileset - lets you change tile symbol or metadata
            </div>
            <div>
              left-click - selects tile 
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App renders tool buttons 1`] = `
<div
  class="tool_wrapper"
  id="toolButtonsWrapper"
>
  <input
    checked=""
    class="hidden"
    id="tool0"
    name="tool"
    type="radio"
    value="0"
  />
  <label
    class="menu"
    data-value="0"
    for="tool0"
    title="paint tiles"
  >
    <div
      id="flipBrushIndicator"
    >
      🖌️
    </div>
    <div
      class="dropdown"
    >
      <div
        class="item nohover"
      >
        Brush tool options
      </div>
      <div
        class="item"
      >
        <label
          class=""
          for="toggleFlipX"
        >
          Flip tile on x
        </label>
        <input
          id="toggleFlipX"
          style="display: none;"
          type="checkbox"
        />
        <label
          class="toggleFlipX"
        />
      </div>
    </div>
  </label>
  <input
    class="hidden"
    id="tool1"
    name="tool"
    type="radio"
    value="1"
  />
  <label
    data-value="1"
    for="tool1"
    title="erase tiles"
  >
    🗑️
  </label>
  <input
    class="hidden"
    id="tool2"
    name="tool"
    type="radio"
    value="2"
  />
  <label
    data-value="2"
    for="tool2"
    title="pan"
  >
    ✋
  </label>
  <input
    class="hidden"
    id="tool3"
    name="tool"
    type="radio"
    value="3"
  />
  <label
    data-value="3"
    for="tool3"
    title="pick tile"
  >
    🎨
  </label>
  <input
    class="hidden"
    id="tool4"
    name="tool"
    type="radio"
    value="4"
  />
  <label
    data-value="4"
    for="tool4"
    title="random from selected"
  >
    🎲
  </label>
  <input
    class="hidden"
    id="tool5"
    name="tool"
    type="radio"
    value="5"
  />
  <label
    data-value="5"
    for="tool5"
    title="fill on layer"
  >
    🌈
  </label>
</div>
`;
